### DISCUSSION TITLE
Create fig generator plugin

### DISCUSSION THREAD

**Discussion by AllanOricil** [oclif/oclif#1197] [Ideas]
@mshanemc @RodEsp @WillieRuemmele @iowillhoit  

https://github.com/forcedotcom/cli/discussions/2514

oclif could accept a new optional boolean flag, called `--fig`, that automatically generates a [fig.io](https://fig.io/)plugin at the end of the build when its value is set to`true`. 

Fig.io is extremely useful for these CLIs that have a huge list of commands.

Since fig.io isn't working on Windows yet, this new flag could be added on the following commands:

- `oclif pack:deb --fig` 
- `oclif pack:mac --fig`

**Comment by RodEsp** [comment]
Hi @AllanOricil, that does sound really cool. It would be great if oclif provided Fig support out-of-the-box.

I am no longer affiliated with oclif though, so I'll let the current maintainers decide if that's a direction they want to take and respond to you appropriately.

**Comment by cristiand391** [comment]
closing this since fig is no longer available
